spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RootController.java

```diff
@@ -74,58 +81,48 @@ public RootController(EntityLinks entityLinks) {
+			resourceSupport.add(unescapeTemplateVariables(entityLinks.linkFor(StepExecutionResource.class, "{jobExecutionId}").slash("{stepId}").withRel("jobs/executions/execution/steps/step")));
+			resourceSupport.add(unescapeTemplateVariables(entityLinks.linkFor(StepExecutionProgressInfoResource.class, "{jobExecutionId}").slash("{stepId}").slash("progress").withRel("jobs/executions/execution/steps/step/progress")));
+			taskTemplated = entityLinks.linkToCollectionResource(JobInstanceResource.class).getHref() + "{?name}";
+			resourceSupport.add(new Link(taskTemplated).withRel("jobs/instances/name"));
+			resourceSupport.add(unescapeTemplateVariables(entityLinks.linkToSingleResource(JobInstanceResource.class, "{id}").withRel("jobs/instances/instance")));
+		}
+		if (featuresProperties.isAnalyticsEnabled()) {
+			resourceSupport.add(entityLinks.linkToCollectionResource(CounterResource.class).withRel("counters"));
+			resourceSupport.add(unescapeTemplateVariables(entityLinks.linkToSingleResource(CounterResource.class, "{name}").withRel("counters/counter")));
+			resourceSupport.add(entityLinks.linkToCollectionResource(FieldValueCounterResource.class).withRel("field-value-counters"));
+			resourceSupport.add(unescapeTemplateVariables(entityLinks.linkToSingleResource(FieldValueCounterResource.class, "{name}").withRel("field-value-counters/counter")));
+			resourceSupport.add(
+					entityLinks.linkToCollectionResource(AggregateCounterResource.class).withRel("aggregate-counters"));
+			resourceSupport.add(unescapeTemplateVariables(entityLinks
+					.linkToSingleResource(AggregateCounterResource.class, "{name}").withRel("aggregate-counters/counter")));
+		}
 		resourceSupport.add(entityLinks.linkToCollectionResource(AppRegistrationResource.class).withRel("apps"));
-
-		resourceSupport.add(
-				entityLinks.linkToCollectionResource(AggregateCounterResource.class).withRel("aggregate-counters"));
-		resourceSupport.add(unescapeTemplateVariables(entityLinks
-				.linkToSingleResource(AggregateCounterResource.class, "{name}").withRel("aggregate-counters/counter")));
-
-		resourceSupport.add(entityLinks.linkToCollectionResource(AppStatusResource.class).withRel("runtime/apps"));
-		resourceSupport.add(unescapeTemplateVariables(entityLinks.linkForSingleResource(AppStatusResource.class, "{appId}").withRel("runtime/apps/app")));
-
-		resourceSupport.add(unescapeTemplateVariables(entityLinks.linkFor(AppInstanceStatusResource.class, UriComponents.UriTemplateVariables.SKIP_VALUE).withRel("runtime/apps/instances")));
-
-		templated = entityLinks.linkFor(CompletionProposalsResource.class).withSelfRel().getHref() + ("/stream{?start,detailLevel}");
-		resourceSupport.add(new Link(templated).withRel("completions/stream"));
-
+		String completionStreamTemplated = entityLinks.linkFor(CompletionProposalsResource.class).withSelfRel().getHref() + ("/stream{?start,detailLevel}");
+		resourceSupport.add(new Link(completionStreamTemplated).withRel("completions/stream"));
```

Comment:
I thought about it. But I don't see any dependencies from Stream for this completion provider. Moreover, I think this is needed for `tasks` commands as well. Hence, I kept it as common.